### DISCUSSION TITLE
Authentication using the artifacts-api

### DISCUSSION THREAD

**Discussion by pefernan** [redhat-developer/app-services-sdk-js#383] [Q&A]
Hi, I'm building a TS package that has to interact with different Apicurio registries to get it's artifacts and contents to provide some autocompletion capabilities to our VsCode extension. Later this could be part of a standalone React app (using monaco editor).

For now we were using axios to fetch the registry artifacts. To handle authentication, in VS Code we are using the RH Authentication extension so we can get the `accessToken` and set it as a Bearer token.

I'm trying your Artifact Api's since they provide the same features we need + diferent authentication mechanisms that we may need in the future. But it seems that Artifacts API isn't using the `accessToken` I pass in the config. What should be the right way to use them with Auth? Noticed in the code that the auth settings are not used there..

Thanks!

**Comment by wtrocki** [comment]
All should work. We use that SDK in numerous places. For instance API you might also consider using upstream SDK:
https://github.com/Apicurio/apicurio-registry-client-sdk-js

What is `RH Authentication extension` ?

**Reply by pefernan** [reply]
@wtrocki [RH Authentication extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-redhat-account&ssr=false#qna) is a VSCode extension that provides Red Hat account authentication  for other vscode extensions.

Going to try the upstream but I've been trying it like:
```
const token = ...;
const serviceRegistryUrl = ...;
const artifactsApi:ArtifactsApi = new ArtifactsApi(new Configuration({
    accessToken: token,
    basePath: serviceRegistryUrl,
  }));

const response = await artifactsApi.searchArtifacts();
```

And I'm getting 401 unauthorized, but if I manually add the token in baseOptions header, like:
```
const token = ...;
const serviceRegistryUrl = ...;
const artifactsApi:ArtifactsApi = new ArtifactsApi(new Configuration({
    basePath: serviceRegistryUrl,
    baseOptions: {
      headers: {
        "Authorization": "Bearer " + token
      }
    }
  }));

const response = await artifactsApi.searchArtifacts();
```
it does work fine,.So I was just wondering if passing the `accessToken` on the config should be enough.

**Reply by pefernan** [reply]
@wtrocki just to confirm, the situation is the same in upstream, taking a look at the code it only adds the headers based on headers passed on the config `baseOptions`, not in the `accessToken` property (https://github.com/Apicurio/apicurio-registry-client-sdk-js/blob/main/src/api.ts#L3668)

Actually the `setBearerAuthToObject` is the function that actually sets the `accessToken` in to the `Authorization` entry in the header which is not called there.

**Reply by wtrocki** [reply]
Ah.. I know the problem now. Both upstream and product use basic auth so workaround is to use header like in example. See https://github.com/Apicurio/apicurio-registry/issues/2369 for more details

**Reply by pefernan** [reply]
@wtrocki thank you for your response. Will continue adding the header like that.
